xbmc/cores/RetroPlayer/RetroPlayer.h

```diff
@@ -98,7 +101,7 @@ class CRetroPlayer : public IPlayer,
   void OnSpeedChange(double newSpeed);
 
   // Playback functions
-  void CreatePlayback(bool bRestoreState);
+  void CreatePlayback(bool bRestoreState, std::string savestatePath);
```

Comment:
Generally we pass strings by const reference (`const std::string&`) to save a copy (though sometimes it's elided by the compiler).

You'll notice above that `GetPlayingGame()` is an exception to this and returns a `std::string` instead of a `const std::string&`. That's because it's an override of a virtual function, and virtual functions shouldn't restrict how implementers manage memory. Recall that if we return `const std::string&`, the memory it references has to be valid the life of the object.